packages/esm-appointments-app/src/appointments/forms/create-edit-form/appointments-form.component.tsx

```diff
@@ -159,15 +174,21 @@ const AppointmentForm: React.FC<AppointmentFormProps> = ({ appointment, patientU
         labelText={t('selectLocation', 'Select a location')}
         id="location"
         invalidText="Required"
-        value={patientAppointment.locationUuid}
-        className={styles.inputContainer}
-        onChange={(event) => setPatientAppointment({ ...patientAppointment, locationUuid: event.target.value })}>
-        {locations?.length > 0 &&
+        value={selectedLocation}
+        defaultSelected={selectedLocation}
+        onChange={(event) => setSelectedLocation(event.target.value)}>
+        {!selectedLocation ? <SelectItem text={t('selectOption', 'Select an option')} value="" /> : null}
+        {!isEmpty(defaultFacility) ? (
+          <SelectItem key={defaultFacility?.uuid} text={defaultFacility?.display} value={defaultFacility?.uuid}>
+            {defaultFacility?.display}
+          </SelectItem>
+        ) : locations?.length > 0 ? (
           locations.map((location) => (
             <SelectItem key={location.uuid} text={location.display} value={location.uuid}>
               {location.display}
             </SelectItem>
-          ))}
+          ))
+        ) : null}
```

Comment:
We can enhance this by extracting this into its own component

```
const SelectOptionComponent = ({ selectedLocation, defaultFacility, locations }) => {
const { t }  = useTranslation();
  if (!selectedLocation) {
    return <SelectItem text={t('selectOption', 'Select an option')} value="" />;
  }

  if (defaultFacility && Object.keys(defaultFacility).length !== 0) {
    return (
      <SelectItem key={defaultFacility?.uuid} text={defaultFacility?.display} value={defaultFacility?.uuid}>
        {defaultFacility?.display}
      </SelectItem>
    );
  }

  if (locations && locations.length > 0) {
    return locations.map((location) => (
      <SelectItem key={location.uuid} text={location.display} value={location.uuid}>
        {location.display}
      </SelectItem>
    ));
  }

  return null;
};
```

then just render it as a child of ``` <SelectOptionComponent {...props}/> </> </Select>```